src/main/java/rx/Observable.java

```diff
@@ -9305,5 +9306,263 @@ public void call(Subscriber<? super T> observer) {
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public static <T, S> Observable<T> create(Action1<ProducerState<T, S>> onRequest) {
+        return create(onRequest, (Func1)UtilityFunctions.returnNull(), Actions.empty()); // type inference shortcomings
+    }
+
+    /**
+     * @warn javadoc
+     */
+    @Experimental
+    public static <T, S> Observable<T> create(Action1<ProducerState<T, S>> onRequest, Func1<Subscriber<? super T>, ? extends S> createState) {
+        return create(onRequest, createState, Actions.empty());
+    }
+
+    /**
+     * @warn javadoc
+     */
+    @Experimental
+    public static <T, S> Observable<T> create(final Action1<ProducerState<T, S>> onRequest, final Func1<Subscriber<? super T>, ? extends S> createState, final Action1<? super S> destroyState) {
+        return create(new OnSubscribe<T>() {
+            @Override
+            public void call(Subscriber<? super T> child) {
+                CallbackProducer<T, S> cp = new CallbackProducer<T, S>(onRequest, createState, destroyState);
+                cp.init(child);
+            }
+        });
+    }
 
+    @Experimental
+    public static <T, S> Observable<T> create(final Func0<AbstractProducer<T, S>> createProducer) {
+        return create(new OnSubscribe<T>() {
+            @Override
+            public void call(Subscriber<? super T> child) {
+                AbstractProducer<T, S> ap = createProducer.call();
+                ap.init(child);
+            }
+        });
+    }
+    
+    /**
+     * Represents the state of the producer.
+     *
+     * @param <T> the emitted value type
+     * @param <S> the user-specified state's type
+     */
+    @Experimental
+    public interface ProducerState<T, S> extends Subscription {
+        /**
+         * Returns the user-created state (may be null).
+         * @return the user-created state (may be null)
+         */
+        S state();
+        /**
+         * Returns the target subscriber.
+         * @return the target subscriber
+         */
+        Subscriber<? super T> subscriber();
+        /**
+         * Returns the current requested amount, Long.MAX_VALUE for essentially unbounded or a negative value if the producer is unsubscribed.
+         * @return the current requested amount
+         */
+        long requested();
+        /**
+         * Indicate the number of items produced and adjust the requested count accordingly.
+         * @param n the number of items produced
+         * @return the new requested amount, negative if the producer is unsubscribed
+         * @throws IllegalArgumentException if n is zero or negative
+         * @throws IllegalStateException if n is greater than the current requested amount
+         */
+        long produced(long n);
+    }
+    
+    /**
+     * Abstract base class to help implement a producer.
+     *
+     * @param <T> the type of the values emitted
+     * @param <S> the associated custom resource type
+     */
+    @Experimental
```

Comment:
Why does this abstract class need to be public? It feels like a complicated implementation detail and I'm wary of adding it to the public API.